packages/ui/src/components/StatusLegend/StatusLegend.tsx

```diff
@@ -1,18 +1,34 @@
 import React from 'react';
-import { useTranslation } from 'react-i18next';
 import { queueStatsStatusList } from '../../constants/queue-stats-status';
-import { toCamelCase } from '../../utils/toCamelCase';
 import s from './StatusLegend.module.css';
+import { NavLink, useLocation } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { toCamelCase } from '../../utils/toCamelCase';
+import cn from 'clsx';
 
 export const StatusLegend = () => {
   const { t } = useTranslation();
+  const { search } = useLocation();
+
   return (
     <ul className={s.legend}>
-      {queueStatsStatusList.map((status) => (
-        <li key={status} className={s[toCamelCase(status)]}>
-          {t(`QUEUE.STATUS.${status.toUpperCase()}`)}
-        </li>
-      ))}
+      {queueStatsStatusList.map((status) => {
+        const displayStatus = t(`QUEUE.STATUS.${status.toUpperCase()}`).toLocaleUpperCase();
+
+        const query = new URLSearchParams(search);
+
+        return (<li key={status} className={cn(s[toCamelCase(status)], {
+          [s.isSelected]: query.get('status') === status,
+        })}>
+          <NavLink
+            to={`/?status=${status}`}
+            className={s[toCamelCase(status)]}
```

Comment:
There is `status` on the parent `li`, why do we need here?

Reply:
I believe that can go as well - that was before I moved the logic out to the `<li>` so it's redundant indeed 👍